sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs

```diff
@@ -49,6 +50,9 @@ public DiagnosticScopeFactory(string clientNamespace, string? resourceProviderNa
 
         public bool IsActivityEnabled { get; }
 
+#if !NET5_0
```

Comment:
If we remove the ifedfs on the DiagnosticScope then we can remove it here as well.